source/numerics/data_out.cc

```diff
@@ -62,31 +62,39 @@ namespace internal
 template <int dim, class DH>
 void
 DataOut<dim,DH>::
-build_one_patch (const std::pair<cell_iterator, unsigned int> *cell_and_index,
-                 internal::DataOut::ParallelData<DH::dimension, DH::space_dimension> &data,
-                 DataOutBase::Patch<DH::dimension, DH::space_dimension> &patch,
-                 const CurvedCellRegion curved_cell_region,
+build_one_patch (const std::pair<cell_iterator, unsigned int>                         *cell_and_index,
+                 internal::DataOut::ParallelData<DH::dimension, DH::space_dimension>  &scratch_data,
+                 const unsigned int                                                    n_subdivisions,
+                 const unsigned int                                                    n_datasets,
+                 const CurvedCellRegion                                                curved_cell_region,
                  std::vector<DataOutBase::Patch<DH::dimension, DH::space_dimension> > &patches)
 {
+  // first create the output object that we will write into
+  ::dealii::DataOutBase::Patch<DH::dimension, DH::space_dimension> patch;
+  patch.n_subdivisions = n_subdivisions;
+  patch.data.reinit (n_datasets,
+                     Utilities::fixed_power<DH::dimension>(n_subdivisions+1));
```

Comment:
Why do you initialize patch.data here? Isn't line 120/131 in the new version taking care of this?